plugins/modules/k8s_service.py

```diff
@@ -195,7 +196,7 @@ def merge_dicts(x, y):
             if isinstance(x[k], dict) and isinstance(y[k], dict):
                 yield (k, dict(merge_dicts(x[k], y[k])))
             else:
-                yield (k, y[k])
+                yield (k, x[k] if x[k] else y[k])
```

Comment:
This would be a breaking change, wouldn't it? I'm also not sure what bug it would be addressing.

Reply:
This is not a breaking change, the module is not working as expected. Here we are trying to merge the definition built using modules parameters and the one from the ``src`` parameter.
Supposed the service port is not set on module parameter, then we are going to merge

(definition from ``src`` parameter)
```
(...)
spec:
   port: xxx
```

definition from module parameters
```
(...)
spec:
   port: None
```
The result will be 
```
(...)
spec:
   port: None
```
The API will complain when trying to create the service, as the port is mandatory.
This fix simply adds a precedence on the value found from the ``src`` option

Reply:
Could you add a bugfix section to the changelog to document this? The change here would need to be `y[k] if y[k] else x[k]`, though. The module params currently take precedence over the resource definition.